### PR TITLE
feat: network value component for re-designed confirmation pages

### LINKED ISSUE #11810: feat: create network value type field
Create value component to display network:

Image

Figma: https://www.figma.com/design/CeAIcptzeaI5ciTpYBaiqr/Confirmations-DS?node-id=4314-18587&node-type=canvas&t=sbkMPvNEVXqdfJec-0

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Adding network value component for use in re-designed confirmation pages.

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/11810

## **Manual testing steps**

1. Enable re-designs locally
2. Go to test dapp
3. Open personal sign page
4. Navigate to network header info section

## **Screenshots/Recordings**
Screenshot 2024-10-16 at 3 16 43 PM

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/confirmations/components/Confirm/Info/PersonalSign/PersonalSign.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ const PersonalSign = () => {
           label={strings('confirm.request_from')}
           tooltip={strings('confirm.personal_sign_tooltip')}
         >
-          <InfoURL url={approvalRequest.origin} />
+          <Url url={approvalRequest.origin} />
```
Comment (jpuri) [inline app/components/Views/confirmations/components/Confirm/Info/PersonalSign/PersonalSign.tsx:26]: I renamed `InfoURL` to `Url`.

File: app/components/Views/confirmations/components/UI/InfoRow/InfoValue/Network/Network.test.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          nickname: 'Ethereum Mainnet',
+          ticker: 'ETH',
+          chainId: CHAIN_IDS.MAINNET,
+        }),
+      },
+    },
+  },
+};
+
+describe('URL', () => {
```
Comment (OGPoyraz) [inline app/components/Views/confirmations/components/UI/InfoRow/InfoValue/Network/Network.test.tsx:25]:
```suggestion
describe('Network', () => {
```

File: app/components/Views/confirmations/components/UI/InfoRow/InfoValue/Network/Network.test.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  it('should display url as expected', async () => {
+    const container = renderWithProvider(
+      <Network chainId={CHAIN_IDS.MAINNET} />,
+      { state: mockInitialState },
+    );
+    expect(container).toMatchSnapshot();
```
Comment (OGPoyraz) [inline app/components/Views/confirmations/components/UI/InfoRow/InfoValue/Network/Network.test.tsx:31]: Can we either expect URL or change this test title to `matches snapshot`?

File: app/components/Views/confirmations/hooks/useNetworkInfo.test.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  },
+};
+
+jest.mock('react-redux', () => ({
+  ...jest.requireActual('react-redux'),
+  useSelector: (fn: (state: DeepPartial<RootState>) => unknown) =>
+    fn(mockInitialState),
+}));
+
+describe('useAccountInfo', () => {
```
Comment (OGPoyraz) [inline app/components/Views/confirmations/hooks/useNetworkInfo.test.ts:34]:
```suggestion
describe('useNetworkInfo', () => {
```

**Review by OGPoyraz — DISMISSED** [review]

File: app/components/Views/confirmations/components/UI/InfoRow/InfoValue/Network/Network.test.tsx (hunk 3 above)
Comment (jpuri) [reply app/components/Views/confirmations/components/UI/InfoRow/InfoValue/Network/Network.test.tsx:31]: done

**Review by OGPoyraz — APPROVED** [review]

**Pull request merged**